catalog/spatial/geocoding/spatial-geocoding-lucene/src/main/java/org/codice/ddf/spatial/geocoding/query/GazetteerQueryCatalog.java

```diff
@@ -0,0 +1,290 @@
+      throw new GeoEntryQueryException(ERROR_MESSAGE, e);
+    }
+
+    return queryResponse
+        .getResults()
+        .stream()
+        .map(Result::getMetacard)
+        .map(this::transformMetacardToGeoEntry)
+        .filter(Objects::nonNull)
+        .collect(Collectors.toList());
+  }
+
+  private GeoEntry transformMetacardToGeoEntry(Metacard metacard) {
+    GeoEntry.Builder geoEntryBuilder = new GeoEntry.Builder();
+    String featureCode =
+        getStringAttributeFromMetacard(metacard, GeoEntryMetacardType.FEATURE_CODE_ATTRIBUTE_NAME);
+    if (StringUtils.isNotBlank(featureCode)) {
+      geoEntryBuilder.featureCode(featureCode);
+    }
+
+    String countryCode = getStringAttributeFromMetacard(metacard, Location.COUNTRY_CODE);
+    if (StringUtils.isNotBlank(countryCode)) {
+      geoEntryBuilder.countryCode(countryCode);
+    }
+
+    String name = getStringAttributeFromMetacard(metacard, Core.TITLE);
+    if (StringUtils.isNotBlank(name)) {
+      geoEntryBuilder.name(name);
+    }
+
+    Attribute attribute = metacard.getAttribute(GeoEntryMetacardType.POPULATION_ATTRIBUTE_NAME);
+    if (attribute != null && attribute.getValue() instanceof Long) {
+      Long population = (Long) attribute.getValue();
+      geoEntryBuilder.population(population);
+    } else {
+      LOGGER.debug("GeoEntry metacard does not contain population attribute.");
+    }
+
+    String location = getStringAttributeFromMetacard(metacard, Core.LOCATION);
+    if (StringUtils.isNotBlank(location)) {
+      try {
+        Geometry geometry = WKT_READER_THREAD_LOCAL.get().read(location);
+        Coordinate coordinate = geometry.getCoordinate();
+        Double lat = coordinate.y;
+        Double lon = coordinate.x;
+        geoEntryBuilder.latitude(lat);
+        geoEntryBuilder.longitude(lon);
+      } catch (com.vividsolutions.jts.io.ParseException e) {
+        LOGGER.debug("GeoEntry metacard does not contain location attribute.");
+      }
+    }
+    return geoEntryBuilder.build();
+  }
+
+  private String getStringAttributeFromMetacard(Metacard metacard, String attributeName) {
+    Attribute attribute = metacard.getAttribute(attributeName);
+    if (attribute != null) {
+      Serializable serializable = attribute.getValue();
+      if (serializable instanceof String) {
+        return (String) serializable;
+      }
+    }
+    return null;
+  }
+
+  @Override
+  public List<NearbyLocation> getNearestCities(String location, int radiusInKm, int maxResults)
+      throws ParseException, GeoEntryQueryException {
+    Filter featureCodeFilter = filterBuilder.anyOf(featureCodeFilters);
+    int radiusInMeters = radiusInKm * KM_TO_M;
+
+    Filter textFilter =
+        filterBuilder.attribute(Core.LOCATION).withinBuffer().wkt(location, radiusInMeters);
+
+    Filter queryFilter = filterBuilder.allOf(featureCodeFilter, tagFilter, textFilter);
+    Query query = new QueryImpl(queryFilter, 1, maxResults, SortBy.NATURAL_ORDER, false, 10000L);
```

Comment:
Should `1000L` be a constant or configurable?

Reply:
made into a constant; in the future we can make it configurable if need be